### PR TITLE
Allow `raw` parameter to accept callable like body elements

### PR DESCRIPTION
**Pull request by ramast**

### CODE REVIEW

File: doc/source/response.rst [hunk 1]
```diff
@@ -70,6 +70,9 @@ Dynamic Response
 ================
 
 A callback can be provided in place of any of the body elements.
+raw attribute also accepts callable that returns HTTPResponse.
+The HTTPResponse should have `preload_content=False` or it may not work properly.
```
Comment (jamielennox) [inline doc/source/response.rst:74]: does it make sense to just always set `raw.preload_content = False` on the raw object? I forget what would happen if this isn't set?

Comment (jamielennox) [reply doc/source/response.rst:74]: `enforce_content_length = False` should probably always be set as well?

**Review by jamielennox — COMMENTED** [review]
I really don't remember why this wasn't available for raw - makes sense for it to be there.

File: doc/source/response.rst (hunk 1 above)
Comment (ramast) [reply doc/source/response.rst:74]: I think we leave them as is to avoid breaking backward compatibility.

Comment (jamielennox) [reply doc/source/response.rst:74]: ok, so i can't set `preload_content` as that is acted upon in the constructor. Setting it afterwards has no impact.

For `enforce_content_length` i think it should probably be false - but the whole point of using `raw` is that the user provides this `HTTPResponse` object so i guess we shouldn't do anything that changes the values they set.

File: requests_mock/response.py [hunk 2]
```diff
@@ -172,20 +172,20 @@ def create_response(request, **kwargs):
...
+            raise TypeError('Text should be string data')
+
+        if json is not None:
+            encoder = kwargs.pop('json_encoder', None) or jsonutils.JSONEncoder
+            text = jsonutils.dumps(json, cls=encoder)
+        if text is not None:
+            encoding = get_encoding_from_headers(headers) or 'utf-8'
+            content = text.encode(encoding)
+        if content is not None:
+            body = _IOReader(content)
```
Comment (jamielennox) [inline requests_mock/response.py:188]: It's not right or wrong - but I don't want to change this block for this purpose.

The whole logic here kind of falls through to the next step based on what's been set and whilst we could reorganize it a little for efficiency it's true for all these statements, not just raw. 

Unless i'm missing something please revert this bit.

**Review by jamielennox — CHANGES_REQUESTED** [review]
Ok, lets leave the logic of the `if not raw` piece. If you want to tackle it let's do it elsewhere, but i'll merge the raw CB.

**Comment by ramast** [comment]
I will not be able to do more work for this feature. partially because I've since moved to other projects that don't need this feature and partially because I forgot the code.

The  pull request is yours, please do with it what you see fit.

**Comment by jamielennox** [comment]
I understand it's pretty old. Thanks for letting me know and I'll pick it up. Apologies again for the delay.

**Comment by ramast** [comment]
of course no need for apology.
Me and everyone who use your library appreciate the time and effort you put into this project

**Comment by jamielennox** [comment]
Moved to #249 as i can't push to your master branch.

**Pull request closed without merging**